section/ontology.md

```diff
@@ -413,3 +413,238 @@ ex:leftPadMapping
+properties used for this are <a href="#fnoc-mapFrom">mapFrom</a> and <a href="#fnoc-mapTo">mapTo</a> . Alternatively, a 
+CompositionMapping can link to a constant term via <a href="#fnoc-mapFromTerm">mapFromTerm</a> instead of 
+<a href="#fnoc-mapFrom">mapFrom</a>.
+
+#### `fnoc:compositionMappingSource` {#fnoc-compositionMappingSource}
+
+The source of a compositionMapping, defining the function and parameter or output that is mapped to the `target`.
+
+#### `fnoc:compositionMappingTarget` {#fnoc-compositionMappingTarget}
+
+The target of a compositionMapping, defining the function and parameter or output that the `source` is mapped to.
+
+
+##### `fnoc:composedOf` {#fnoc-composedOf}
+
+**Domain** <a href="#fnoc-composition">fnoc:Composition</a>
+
+**Range** <a href="#fnoc-compositionMapping">fnoc:CompositionMapping</a> 
+
+##### `fnoc:applies` {#fnoc-applies}
+
+**Domain** <a href="#fn-function">fno:Function</a>
+
+**Range** <a href="#fn-function">fno:Function</a> 
+
+##### `fnoc:constituentFunction` {#fnoc-constituentFunction}
+
+**Domain** <a href="#fnoc-compositionMappingSource">fnoc:CompositionMappingSource</a> or <a href="#fnoc-compositionMappingSource">fnoc:CompositionMappingTarget</a>
+
+**Range** <a href="#fn-function">fno:Function</a>
+
+##### `fnoc:functionOutput` {#fnoc-functionOutput}
+
+**Domain** <a href="#fnoc-compositionMappingSource">fnoc:CompositionMappingSource</a> or <a href="#fnoc-compositionMappingSource">fnoc:CompositionMappingTarget</a>
+
+**Range** <a href="#fn-output">fno:Output</a>
+
+##### `fnoc:functionParameter` {#fnoc-functionParameter}
+
+**Domain** <a href="#fnoc-compositionMappingSource">fnoc:CompositionMappingSource</a> or <a href="#fnoc-compositionMappingSource">fnoc:CompositionMappingTarget</a>
+
+**Range** <a href="#fn-output">fno:Parameter</a>
+
+##### `fnoc:mapFrom` {#fnoc-mapFrom}
+
+**Domain** <a href="#fnoc-compositionMapping">fnoc:CompositionMapping</a>
+
+**Range** <a href="#fnoc-compositionMappingSource">fnoc:CompositionMappingSource</a> 
+
+##### `fnoc:mapFromTerm` {#fnoc-mapFromTerm}
+
+**Domain** <a href="#fnoc-compositionMapping">fnoc:CompositionMapping</a>
+
+**Range** (any) 
+
+Allows for using constants in a <a>composition</a>.
+
+
+##### `fnoc:mapTo` {#fnoc-mapTo}
+
+**Domain** <a href="#fnoc-compositionMapping">fnoc:CompositionMapping</a>
+
+**Range** <a href="#fnoc-compositionMappingTarget">fnoc:CompositionMappingTarget</a> 
+
+### Partial Function Application {#partialFunctionApplication}
+
+In the following example, we define the function `ex:add10` as a partial application of `ex:sumFunction` by providing 
+the constant value `10` for the parameter `ex:intParameterA`. The parameter value is specified using the associated 
+<a href="#fn-predicate">fno:predicate</a> `ex:startValue`. 
+
+```turtle "example": " " 
+ex:add10 a fnoc:PartiallyAppliedFunction;
+    fnoc:partiallyApplies :sumFunction ;
+    ex:startValue 10 .  
```

Comment:
Hmmm, I'm not really a fan of directly applying the predicate to the function, that's kind of 'crossing boundaries'. Would something like below be an option?

```turtle
ex:add10 a fnoc:PartiallyAppliedFunction;
    fnoc:partiallyApplies :sumFunction ;
    fnoc:partialApplication [
      fnoc:mapFromTerm 10 ;
      fnoc:mapToParameter ex:inParameterA
    ] .
```

Reply:
Yes, that's good. Still concise compared to the full composition equivalent, and ontologically clean.

Reply:
On second thought, wouldn't it be easier to re-use the composedOf property?
```
ex:add10 a fnoc:PartiallyAppliedFunction;
    fnoc:partiallyApplies :sumFunction ;
    fnoc:composedOf [
      fnoc:mapFromTerm 10 ;
      fnoc:mapToParameter ex:inParameterA
    ] .
```

Reply:
I'm fine with your latest suggestion. It might not be perfect yet, but it's at least something we can start from :)